dashboard/test/controllers/api/v1/peer_review_submissions_controller_test.rb

```diff
@@ -3,7 +3,12 @@
 class Api::V1::PeerReviewSubmissionsControllerTest < ActionController::TestCase
   self.use_transactional_test_case = true
 
-  setup_all do
+  setup do
+    @plc_reviewer = create :plc_reviewer
+    sign_in(@plc_reviewer)
+  end
+
+  def common_scenario
```

Comment:
This function does a fair amount of creating new records and make them as instance variables. Why is it better to take it out of setup_all?

Reply:
Because only four of the test cases in this file depend on this complex setup.  Rather than have the other tests inherit it via `setup_all` and ignore it, I'm making the four that need this scenario call for it explicitly.  While that is technically less efficient than creating these records in `setup_all`, it means there will be less state to reason about when reading and writing tests here in the future.

Reply:
Do we have to concern about data conflict when tests are running in parallel? Each test will create different data but they assign them to the same instance variables. Or we implicitly rely on database isolation when tests are executed on `test` environment?

Reply:
We use https://github.com/grosser/parallel_tests which runs each group of tests in its own process with its own database, so tests running simultaneously will not share instance variables or database rows.

Reply:
That what I thought, but just want to confirm. And these tests run sequentially on development and Drone, right?

Reply:
Correct.